packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -270,13 +269,16 @@ export class DataStores implements IDisposable {
     }
 
     public async getDataStore(id: string, wait: boolean): Promise<IFluidDataStoreChannel> {
-        const deferredContext = this.contexts.prepDeferredContext(id);
+        const existingContext = this.contexts.get(id);
+        if (existingContext !== undefined && !this.contexts.isNotBound(id)) {
```

Comment:
this method has always been subtle. it looks right to me, but unit tests would be great if they don't already exist. it might be easier to test dataStoreContexts, and since this is the only place we use waitForContext it might make sense to move this logic to dataStoreContexts

Reply:
@anthony-murphy - Great point, I'm thinking this method (minus the call to `.realize()`) moves to `DataStoreContexts.getAttached`. Or if `Attached` is not the right adjective for the context for a datastore that exists on the server, what is? `Remoted`? I'm reading through `DataStoreContext.ts` now trying to understand the difference between the different concrete classes...

Reply:
The semantics are complicated, and it doesn't fit neatly with any of our current verb, as when detached container or local it only returns bound contexts, not unbound. But for an attached container if can also be used for remote context. Personally, i would avoid referencing in either. I like just calling it waitForContext as it is now, but just putting more implementation in it, and then adding a bunch of comments that describe the behavior

Reply:
Ok gotcha.  Will update.

Reply:
Went with `getBoundOrRemoted` to match `addBoundOrRemoted` (note the added `d` to `Remote` to match the class `RemotedDataStoreContext)